helm/opendistro-es/files/configurer/configurer.sh

```diff
@@ -0,0 +1,205 @@
+    resp=$(curl -X PUT "${es_url}/_template/${index}?create=${only_create_templates}" \
+      -H "Content-Type: application/json" -s \
+      -d@/files/${filename} -k -u "${auth}")
+    acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+    if [ "${acknowledged}" != "true" ]; then
+      if [ $only_create_templates = "true" ] \
+          && echo "${resp}" | grep "already exists" > /dev/null ; then
+        echo "Index template '${index}' already exists, do nothing"
+      else
+        log_error_exit "Failed to create index template for file '${filename}'" "${resp}"
+      fi
+    fi
+  done
+}
+
+setup_policies() {
+  create_update_policy() {
+    update_policies="{{ .Values.configurer.updatePolicies }}"
+
+    update_policy() {
+      policy=$1
+      policy_json=$(curl -X GET "${es_url}/_opendistro/_ism/policies/${policy}" \
+        -H "Content-Type: application/json" -k -s \
+        -u "${auth}")
+      seq_no=$(echo "${policy_json}" | jq -r '._seq_no')
+      primary_term=$(echo "${policy_json}" | jq -r '._primary_term')
+      resp=$(curl -X PUT "${es_url}/_opendistro/_ism/policies/${policy}?if_seq_no=${seq_no}&if_primary_term=${primary_term}" \
+        -H "Content-Type: application/json" -k -s \
+        -d@"/files/${policy}_policy.json" \
+        -u "${auth}")
+      id=$(echo "${resp}" | grep "^{" | jq -r '._id')
+      if [ "${id}" != "${policy}" ]; then
+        log_error_exit "Failed to update policy '${policy}'" "${resp}"
+      fi
+      echo "Updated policy '${policy}'"
+    }
+
+    policy=$1
+    echo "Creating policy '${policy}'"
+    resp=$(curl -X PUT "${es_url}/_opendistro/_ism/policies/${policy}" \
+      -H "Content-Type: application/json" \
+      -d@"/files/${policy}_policy.json" -k -s \
+      -u "${auth}")
+    status=$(echo "${resp}" | grep "^{" | jq -r '.status')
+    id=$(echo "${resp}" | grep "^{" | jq -r '._id')
+    if [ "${status}" = 409 ]; then # policy already exists
+      echo "Policy '${policy}' already exists"
+      if [ "${update_policies}" = "true" ]; then update_policy "${policy}"; fi
+    elif [ "${id}" != "${policy}" ]; then
+      log_error_exit "Unknown response, failed to create policy?" "${resp}"
+    fi
+  }
+
+  index_set_policy() {
+    policy=$1 # assume policy name == index name
+    resp=$(curl -X POST "${es_url}/_opendistro/_ism/add/${policy}" \
+      -H "Content-Type: application/json" \
+      -d '{"policy_id": "'"${policy}"'"}' \
+      -k -s -u "${auth}")
+    failures=$(echo "${resp}" | grep "^{" | jq -r '.failures')
+    if [ "${failures}" = "false" ]; then
+      echo "Set policy '${policy}' on index '${policy}'"
+    elif [ "${failures}" = "true" ]; then
+      updated_indices=$(echo "${resp}" | jq -r '.updated_indices')
+      echo "Index '${policy}' already has a policy '${policy}'"
+      echo "Updated ${updated_indices} indices"
+    else
+      log_error_exit "Unknown response" "${resp}"
+    fi
+  }
```

Comment:
I think that we can skip this step right here. The policy should be assigned to the indexes via the index templates that we create.
```suggestion
```

Reply:
Great! Committed the suggestion and appended some cleanup locally :slightly_smiling_face: